### PR TITLE
set extension for elixir-ls based on OS

### PR DESCRIPTION
**Pull request by erikbye**

### CODE REVIEW

File: alchemist-elixir-ls.el [hunk 1]
```diff
@@ -55,14 +50,10 @@
...
-  (let ((extension (or alchemist-server-extension
-                       (completing-read "Choose the kind of executable that runs on this system: "
-                                        '("sh" "bat")
-                                        nil
-                                        t
-                                        ))))
-    (setq alchemist-server-extension extension)
-    extension))
+  (defvar alchemist-system-extension)
+  (cond ((eq system-type 'gnu/linux) (setq alchemist-system-extension "sh"))
```
Comment (Trevoke) [inline alchemist-elixir-ls.el]: Is it possible at all for a system to be something other than these three? If so we may want to at least raise an error.

The pattern I've used for something like this in the past was like this: https://github.com/Trevoke/sqlup-mode.el/blob/master/sqlup-mode.el#L199

Is yours more e-lispy? Off-hand it seems weird to have the variable definition inside the function definition.

File: alchemist-elixir-ls.el [hunk 2]
```diff
@@ -55,14 +50,10 @@
...
 (defun alchemist--server-extension ()
-  (let ((extension (or alchemist-server-extension
-                       (completing-read "Choose the kind of executable that runs on this system: "
-                                        '("sh" "bat")
-                                        nil
-                                        t
-                                        ))))
-    (setq alchemist-server-extension extension)
-    extension))
+  (defvar alchemist-system-extension)
```
Comment (Trevoke) [inline alchemist-elixir-ls.el]: If the variable is no longer public-facing then we should call it `alchemist--system-extension` to indicate that it is private.

**Review by Trevoke — COMMENTED** [review]
Thank you for the PR!

This is definitely an improvement. I've left two comments/questions about it, let me know your thoughts.

File: alchemist-elixir-ls.el (hunk 1 above)
Comment (erikbye) [reply alchemist-elixir-ls.el]: Possible values for system-type:
```
  ‘gnu’          compiled for a GNU Hurd system.
  ‘gnu/linux’    compiled for a GNU/Linux system.
  ‘gnu/kfreebsd’ compiled for a GNU system with a FreeBSD kernel.
  ‘darwin’       compiled for Darwin (GNU-Darwin, macOS, ...).
  ‘ms-dos’       compiled as an MS-DOS application.
  ‘windows-nt’   compiled as a native W32 application.
  ‘cygwin’       compiled using the Cygwin library.
Anything else (in Emacs 26, the possibilities are: aix, berkeley-unix,
hpux, usg-unix-v) indicates some sort of Unix system.
```
The question is which operating systems does elixir-ls run on, every OS Elixir and Erlang runs on?

More e-lispy? Probably not.

File: alchemist-elixir-ls.el (hunk 2 above)
Comment (erikbye) [reply alchemist-elixir-ls.el]: Sure.

**Comment by erikbye** [comment]
We could just set the extension to .sh by default, and only explicitly set it to .bat if you're on Windows. But, if you ever want to raise an error and let the user know his OS is not supported, I guess this would be a good place to do it, anyway.

Not even sure extension needs to be specified, at least on Windows, this works fine:

```
(defun alchemist--lsp-server-path-for-current-project ()
  `(,(concat alchemist-server-root-path
           "erl"
           (alchemist--server-erlang-version (alchemist-project-root-or-default-dir))
           "/"
           "language_server")))
```
Haven't tested on macOS/Linux, yet.

**Comment by Trevoke** [comment]
Oh wow, I could have sworn I'd merged this in, well, almost a month ago! Sorry about the massive delay. Thank you for the contribution!

**Pull request merged**